### PR TITLE
Support adding DNS entries for CIS service based cluster domain

### PR DESCRIPTION
**Pull request by Prajyot-Parab**
- Support adding DNS entries for CIS service based cluster domain

### CODE REVIEW

**Comment by Prajyot-Parab** [comment]
/cc @yussufsh 
This is critical for the MAC P2 CI

File: ocp.tf [hunk 1]
```diff
@@ -205,6 +205,18 @@ module "install" {
...
 }
 
+module "post_install" {
+  count      = var.ibm_cloud_cis_crn != "" && !var.use_ibm_cloud_services ? 1 : 0
+  source     = "./modules/6_post_install"
+  depends_on = [module.install]
+
+  cluster_domain    = module.nodes.cluster_domain
+  cluster_id        = local.cluster_id
+  bastion_ip        = module.prepare.bastion_ip
```
Comment (yussufsh) [inline ocp.tf:215]: use bastion_vip in case count > 1

File: ocp.tf [hunk 2]
```diff
@@ -205,6 +205,18 @@ module "install" {
...
+  bastion_public_ip = module.prepare.bastion_public_ip
```
Comment (yussufsh) [inline ocp.tf:216]: Use bastion_external_vip in case count > 1

File: modules/6_post_install/cis.tf [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  cis_id = var.ibm_cloud_cis_crn
+  domain = var.cluster_domain
+}
+
+#####################################
+##### Kubernetes 
+#####################################
+resource "ibm_cis_dns_record" "api" {
+  cis_id    = var.ibm_cloud_cis_crn
+  content   = var.bastion_public_ip[0]
```
Comment (yussufsh) [inline modules/6_post_install/cis.tf:35]: variable can be string.

File: modules/6_post_install/cis.tf [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  domain_id = data.ibm_cis_domain.domain.id
+  name      = "api.${var.cluster_id}.${var.cluster_domain}"
+  ttl       = 900
+  type      = "A"
+}
+resource "ibm_cis_dns_record" "api-int" {
+  cis_id    = var.ibm_cloud_cis_crn
+  content   = var.bastion_ip[0]
```
Comment (yussufsh) [inline modules/6_post_install/cis.tf:43]: variable can be string.

File: modules/6_post_install/cis.tf [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  domain_id = data.ibm_cis_domain.domain.id
+  name      = "api-int.${var.cluster_id}.${var.cluster_domain}"
+  ttl       = 900
+  type      = "A"
+}
+resource "ibm_cis_dns_record" "apps" {
+  cis_id    = var.ibm_cloud_cis_crn
+  content   = var.bastion_public_ip[0]
```
Comment (yussufsh) [inline modules/6_post_install/cis.tf:51]: variable can be string.

File: modules/6_post_install/cis.tf [hunk 6]
```diff
@@ -0,0 +1,56 @@
+################################################################
```
Comment (yussufsh) [inline modules/6_post_install/cis.tf:1]: Can you please rename the dir to `8_custom`

**Review by yussufsh — CHANGES_REQUESTED** [review]
@Prajyot-Parab Please rebase to take CIS changes.

File: docs/quickstart.md [hunk 7]
```diff
@@ -136,6 +136,8 @@ Once the deployment is completed successfully, you can safely delete the bootstr
 
 Please skip this section if your `cluster_domain` is one of the online wildcard DNS domains: nip.io, xip.io and sslip.io. Also, not required when using `use_ibm_cloud_services = true`.
 
+>**Note**: If you are using IBM Cloud Internet Services (CIS) for `cluster_domain`, ensure you set `ibm_cloud_cis_crn`.
```
Comment (yussufsh) [inline docs/quickstart.md:139]: Rephrase:
If you just need to add the DNS entries in the CIS domain zone without using IBM Cloud Services eg: VPC Load Balancer, then set `ibm_cloud_cis_crn` with the CIS domain CRN and keep `use_ibm_cloud_services = false`.

File: modules/8_custom/cis.tf [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+  type      = "A"
+}
+resource "ibm_cis_dns_record" "api-int" {
+  cis_id    = var.ibm_cloud_cis_crn
+  content   = var.bastion_ip
+  domain_id = data.ibm_cis_domain.domain.id
+  name      = "api-int.${var.cluster_id}.${var.cluster_domain}"
+  ttl       = 900
+  type      = "A"
+}
```
Comment (yussufsh) [inline modules/8_custom/cis.tf:48]: We do not need internal DNS entry. This will be used from bastion for the cluster operations.

**Review by yussufsh — CHANGES_REQUESTED** [review]

**Review by yussufsh — APPROVED** [review]
/lgtm

**Pull request merged**